crypto_analytics/collection/data_source/crypto_compare.py

```diff
@@ -35,30 +38,41 @@ def fetch(self) -> pd.DataFrame:
         return self.data
 
     def write(self, filepath: str):
-        self.data.to_csv (filepath)
+        self.data.to_csv(filepath)
 
     def get_time(self):
         return self.data['time']
 
-    def _get_ochlv(self, ochlv_type: str, interval: Interval):
-        self.endpoint = self.valid_time_intervals.get(interval)
-        if not self.endpoint:
-            raise ValueError('Interval must be daily, hourly or minute')
+    def get_open(self):
+        return self.data['open']
 
-        ochlv = self.fetch()
-        return ochlv['Data'][ochlv_type]
+    def get_close(self):
+        return self.data['close']
 
-    def get_open(self, interval: Interval):
-        self._get_ochlv('open', interval)
```

Comment:
I think these `get_open` methods should read from self.data so that say when we want both `get_open` and `get_close` we don't have to make two api calls

Reply:
Thats  good idea.  I want to implement a caching strategy eventually anyways so we can overall reduce the number of API calls.  This a good start to that.